### PR TITLE
backend: Make validation dates more strict

### PR DESCRIPTION
**Pull request by hanadikaleel**
#1861

### CODE REVIEW

File: inspire_utils/date.py [hunk 1]
```diff
@@ -99,7 +99,15 @@ def loads(cls, string):
             ValueError: month must be in 1..12
 
         """
-        parts = (int(part) for part in string.split('-'))
+
+        date_parts = string.split('-')
+
+        if len(date_parts) == 2 and len(date_parts[1]) < 2:
+            raise ValueError('Month must be in MM format')
+        if len(date_parts) == 3 and len(date_parts[2]) < 2:
```
Comment (michamos) [inline inspire_utils/date.py:107]: there's another case that you're not catching, where there's a full date but only the month is missing a leading 0, like in `2015-1-15`.

File: inspire_utils/date.py [hunk 2]
```diff
@@ -99,7 +99,17 @@ def loads(cls, string):
             ValueError: month must be in 1..12
 
         """
-        parts = (int(part) for part in string.split('-'))
+
+        date_parts = string.split('-')
+
+        if len(date_parts) == 2 and len(date_parts[1]) < 2:
+            raise ValueError('Month must be in MM format')
+        if len(date_parts) == 3 and len(date_parts[1]) < 2:
```
Comment (michamos) [inline inspire_utils/date.py:107]: alternatively, you could change L105 from `== 2` to `>= 2`.

File: tests/test_date.py [hunk 3]
```diff
@@ -33,6 +33,11 @@
 )
 
 
+def test_loads_validate_dates():
+    with pytest.raises(ValueError):
+        PartialDate.loads('2015-1-1')
+
+
```
Comment (michamos) [inline tests/test_date.py:40]: can you add a test case for the example I gave you?

Comment (hanadikaleel) [reply tests/test_date.py:41]: updated

Comment (michamos) [reply tests/test_date.py:41]: that doesn't work. The first call will raise, so the second one will never be executed. You need to split it into two.

**Pull request merged**